src/main/webapp/app/exam/manage/exercise-groups/exercise-groups.component.ts

```diff
@@ -251,6 +256,43 @@ export class ExerciseGroupsComponent implements OnInit {
         }
     }
 
+    getFilePatternOfExercise(exercise: Exercise) {
+        if (exercise.type === this.exerciseType.FILE_UPLOAD) {
+            return (exercise as FileUploadExercise).filePattern ?? '';
+        } else {
+            return '';
+        }
+    }
+
+    getDiagramTypeOfExercise(exercise: Exercise) {
+        if (exercise.type === this.exerciseType.MODELING) {
+            return (exercise as ModelingExercise).diagramType ?? '';
+        } else {
+            return '';
+        }
+    }
+
+    getQuizQuestionAmountOfExercise(exercise: Exercise) {
+        if (exercise.type === this.exerciseType.QUIZ) {
+            return (exercise as QuizExercise).quizQuestions?.length ?? 0;
+        }
+        return 0;
+    }
+
+    toProgrammingExercise(exercise: Exercise) {
+        if (exercise.type === this.exerciseType.PROGRAMMING) {
```

Comment:
Using such methods can be problematic, because they are much slower to check for Angular's automatic change detection than accessing the attributes directly. We should be careful here and avoid them!